client/state/comments/reducer.js

```diff
@@ -509,6 +530,26 @@ export const counts = ( state = {}, action ) => {
 			);
 			return Object.assign( {}, state, { [ siteId ]: newTotalSiteCounts } );
 		}
+		case COMMENTS_EMPTY_RECEIVE: {
+			const { siteId, status, commentIds } = action;
+
+			if ( ! siteId || ! state[ siteId ] || ! status ) {
+				return state;
+			}
+			const { site: siteCounts } = state[ siteId ];
+
+			const emptiedCommentsCount = commentIds?.length || 0;
+			const newSiteCounts = updateCount( siteCounts, status, -emptiedCommentsCount );
+
+			// Post counts can't be updated here because we don't know the post ID.
+
+			const newTotalSiteCounts = Object.assign(
+				{},
+				state[ siteId ],
+				newSiteCounts && { site: newSiteCounts }
+			);
+			return Object.assign( {}, state, { [ siteId ]: newTotalSiteCounts } );
```

Comment:
Nit: object spread could simplify `Object.assign()` calls a bit:

```suggestion
			const newTotalSiteCounts = Object.assign(
				{},
				state[ siteId ],
				newSiteCounts && { site: newSiteCounts }
			);
			return { ...state, ...{ [ siteId ]: newTotalSiteCounts } };
```

Reply:
Thanks :) I was reusing old code here. Much neater with the spread operator.